hollow/src/main/java/com/netflix/hollow/api/error/IncompatibleSchemaException.java

```diff
@@ -0,0 +1,24 @@
+package com.netflix.hollow.api.error;
+
+/**
+ * An exception thrown when trying to compare two versions of an object with incompatible schema.
+ */
+public class IncompatibleSchemaException extends HollowException {
+    private final String typeName;
+    private final String fieldName;
+
+    public IncompatibleSchemaException(String typeName, String fieldName) {
+        super("No common schema exists for " + typeName + ": field " + fieldName
```

Comment:
It would be good to capture more context if available. For example, types changed.

Reply:
Hm good idea, I'll add some more info.